pkg/controller/topology.go

```diff
@@ -238,37 +268,56 @@ func GenerateAccessibilityRequirements(
 	requisiteTerms = slices.CompactFunc(requisiteTerms, slices.Equal)
 	// TODO (verult) reduce subset duplicate terms (advanced reduction)
 
+	// Filter out empty topology terms to prevent issues with caching and spreading.
+	// An empty term can be generated from an empty TopologySelectorTerm,
+	// but should not be considered for placement.
+	requisiteTerms = filterEmptyTerms(requisiteTerms)
```

Comment:
`toCSITopology` already skips empty terms. Do we still need `filterEmptyTerms` here?